apollo-federation/src/operation/mod.rs

```diff
@@ -66,23 +65,13 @@ pub(crate) use rebase::*;
 
 pub(crate) const TYPENAME_FIELD: Name = name!("__typename");
 
-// Global storage for the counter used to uniquely identify selections
-static NEXT_ID: atomic::AtomicUsize = atomic::AtomicUsize::new(1);
-
 /// Opaque wrapper of the unique selection ID type.
-///
-/// Note that we shouldn't add `derive(Serialize, Deserialize)` to this without changing the types
-/// to be something like UUIDs.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
-// NOTE(@TylerBloom): This feature gate can be removed once the condition in the comment above is
-// met. Note that there are `serde(skip)` statements that should be removed once this is removed.
-#[cfg_attr(feature = "snapshot_tracing", derive(Serialize))]
-pub(crate) struct SelectionId(usize);
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, serde::Serialize)]
+pub(crate) struct SelectionId(Uuid);
```

Comment:
should this change be in a separate PR?

Reply:
This can be moved into a different PR or reverted. My motivation was to make our serialization uniform (i.e. not feature gating anything), and both using `apollo-compiler`'s `serialize` methods and removing feature-gating fall into this. While we aren't serializing to cache, simply implementing `Serialize` would violate what that comment is saying but changing the representation and not cache is still inline.

My goal with making things uniform (and specifically removing the feature-gating) was to minimize what we would need to consider when/if we started to serialize for caching. That said, this is ultimately a small change that can be revisited at that point in time if this change isn't inline with what would be needed.

Reply:
I think the comment is a bit over-eager and it should maybe just generally warn about it being a fleeting value that is not designed to be unique between planning runs

Reply:
Oh, the uniform attributes point makes sense. I think it's okay to not feature gate serialization and warn about deserialization in the comment?

Reply:
>  I think it's okay to not feature gate serialization and warn about deserialization in the comment?

I think this works. We already do not implement `Deserialize` and there are several outstanding questions around what that would look like (there is even a new one in the PR). I'll revert the `uuid` changes, update that comment, and leave a new comment (either at the top of `lib.rs` or by `QueryPlan`) saying that we don't support deserialization yet and we need to answer a few questions before we can.